src/cdk/stepper/step-label.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, TemplateRef, ViewContainerRef} from '@angular/core';
+import {TemplatePortalDirective} from '../portal';
+
+@Directive({
+  selector: '[cdk-step-label]',
+})
+export class CdkStepLabel extends TemplatePortalDirective {
```

Comment:
I think you can do this the same way you did `CdkStep` (just a `TemplateRef` and `ngTemplateOutlet` tabs used the portal stuff because it was written before `ngTemplateOutlet` was a thing